py3status/modules/online_status.py

```diff
@@ -50,17 +55,19 @@ def _connection_present(self):
                                    stdout=fnull, stderr=fnull) == 0
 
     def online_status(self):
+
         response = {
-            'cached_until': self.py3.time_in(self.cache_timeout)
+            'cached_until': self.py3.time_in(self.cache_timeout),
+            'full_text': self.py3.safe_format(self.format, {'state': self.format_offline}),
+            'color': self.py3.COLOR_BAD
         }
```

Comment:
response is calculated and then just dumped if we are connected.  We should only be calculating the stuff we need to.


`py3.safe_format()` is very expensive so we want to only do it when we need to.  At some point I want to re-write the formatter so it is much more efficient but that is another story.

also 'cached_until' is not part of response when we are connected